urnai/sc2/environments/sc2environment.py

```diff
@@ -0,0 +1,134 @@
+from pysc2.env import sc2_env
+from pysc2.env.environment import StepType
+from pysc2.lib import actions, features
+
+from urnai.environments.environment_base import EnvironmentBase
+
+sc2_races = {
+    'terran': sc2_env.Race.terran,
+    'protoss': sc2_env.Race.protoss,
+    'zerg': sc2_env.Race.zerg,
+    'random': sc2_env.Race.random}
+sc2_difficulties = {
+    'very_easy': sc2_env.Difficulty.very_easy,
+    'easy': sc2_env.Difficulty.easy,
+    'medium': sc2_env.Difficulty.medium,
+    'medium_hard': sc2_env.Difficulty.medium_hard,
+    'hard': sc2_env.Difficulty.hard,
+    'harder': sc2_env.Difficulty.harder,
+    'very_hard': sc2_env.Difficulty.very_hard,
+    'cheat_vision': sc2_env.Difficulty.cheat_vision,
+    'cheat_money': sc2_env.Difficulty.cheat_money,
+    'cheat_insane': sc2_env.Difficulty.cheat_insane}
+
+
+class SC2Env(EnvironmentBase):
+    def __init__(
+            self,
+            map_name='Simple64',
+            players=None,
+            player_race='terran',
+            enemy_race='random',
+            difficulty='very_easy',
+            render=False,
+            reset_done=True,
+            spatial_dim=16,
+            step_mul=16,
+            game_steps_per_ep=0,
```

Comment:
What's `ep`?

Reply:
It's an episode. This parameter is used to initialize the `pysc2` environment. Should I rename it to `game_steps_per_episode`?

Reply:
> Should I rename it to game_steps_per_episode?

Yep.